public/controlpanel.php

```diff
@@ -395,22 +395,18 @@ function validateEmail() {
             <?php
             if ($errorCode == 'baddata' || $errorCode == 'generalerror') {
                 echo "<div id=\"warning\">Info: Errors changing your password. Please check and try again!</div>";
-            } else {
-                if ($errorCode == 'badnewpass') {
-                    echo "<div id=\"warning\">Info: Errors changing your password, passwords too short!</div>";
-                } else {
-                    if ($errorCode == 'passinequal') {
-                        echo "<div id=\"warning\">Info: Errors changing your password, new passwords were not identical!</div>";
-                    } else {
-                        if ($errorCode == 'badpassold') {
-                            echo "<div id=\"warning\">Info: Errors changing your password, old password was incorrect!</div>";
-                        } else {
-                            if ($errorCode == 'changepassok') {
-                                echo "<div id=\"warning\">Info: Password changed OK!</div>";
-                            }
-                        }
-                    }
-                }
+            }
+            if ($errorCode == 'badnewpass') {
+                echo "<div id=\"warning\">Info: Errors changing your password, passwords too short!</div>";
+            }
+            if ($errorCode == 'passinequal') {
+                echo "<div id=\"warning\">Info: Errors changing your password, new passwords were not identical!</div>";
+            }
+            if ($errorCode == 'badpassold') {
+                echo "<div id=\"warning\">Info: Errors changing your password, old password was incorrect!</div>";
+            }
+            if ($errorCode == 'changepassok') {
```

Comment:
these should probably be `elseif`s - only one will ever match at a time.

Reply:
while that's true I personally dislike `if/elseif/else` chains. i favor clean and easy to ready constructs over minor performance gains.
i think this should be rewritten to a `match` clause where the message is set, and if, display a html element containing the message.
my goal in this case was to untangle the unnecessary indentations for future refactoring.